monai/metrics/metric.py

```diff
@@ -140,64 +184,103 @@ def reset(self):
         self._synced_tensors = None
         self._synced = False
 
-    def add(self, *data: torch.Tensor):
+    def extend(self, *data) -> None:
         """
-        Add samples to the cumulative buffers.
+        Extend the local buffers with new ("batch-first") data.
+        A buffer will be allocated for each `data` item.
 
         Args:
-            data: list of input tensor, make sure the input data order is always the same in a round.
-                every item of data will be added to the corresponding buffer.
+            data: each item can be a "batch-first" tensor or a list of "channel-first" tensors.
+                they will be concatenated at the 0-th dimension when `get_buffer()` is called.
+        """
+        if self._buffers is None:
+            self._buffers = [[] for _ in data]
+        for b, d in zip(self._buffers, data):
+            # converting to pytorch tensors so that we can use the distributed API
+            d_t: torch.Tensor
+            d_t, *_ = convert_data_type(d, output_type=torch.Tensor, wrap_sequence=True)  # type: ignore
+            try:
+                b.extend([x[0] for x in torch.split(d_t, 1, dim=0)])
```

Comment:
I doult there may be a little performance drop to split the batch into list and stack them in `sync()`, the previous code store `batch-first` Tensor into buffer directly and concat several `batch-first` Tensors in `sync()`.
Maybe you can do some quick performance test if batch size is bigger?

Thanks in advance.

Reply:
Hi Nic, this is to have consistent buffer elements in the implementation, and the base class is to show the concepts. If the performance is an issue, the subclass can provide other implementations.